pkg/util/inplaceupdate/inplace_utils.go

```diff
@@ -0,0 +1,86 @@
+package inplaceupdate
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appspub "github.com/openkruise/kruise/apis/apps/pub"
+	"github.com/openkruise/kruise/pkg/util"
+	"github.com/openkruise/kruise/pkg/util/metrics/update"
+	"github.com/openkruise/kruise/pkg/util/podadapter"
+)
+
+const AnnotationRestartCountInfoKey = "inplace.apps.kruise.io/restart-count-info"
+
+type RestartCountInfo struct {
+	RestartCountBase map[string]int32 `json:"base"`
```

Comment:
is it possible to use the map directly in the "restart-count-info" without the extra `base` field?